src/game/Entities/Pet/Pet.cpp

```diff
@@ -999,6 +999,10 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
     UpdateAllStats();
 
     SetFullHealth();
+    if (GetEntry() == NPC_RISEN_GHOUL)
+    {
+        SetPower(POWER_ENERGY, GetMaxPower(POWER_ENERGY));
```

Comment:
few lines before there is already an `if (GetEntry() == NPC_RISEN_GHOUL)` block, why don't you put that there instead of creating another if block?